Task5/MainSolution/ShopModel/UnitTestsClasses.cs

```diff
@@ -0,0 +1,129 @@
+using NUnit.Framework;
+using Core.Discounts;
+using Core;
+
+namespace ShopModel
+{
+
+    public class Tests
+    {
+
+        [Test]
+        public void TestClassShoppingCartCalculateTotalDiscount_1()
+        {
+            Product[] products =
+            {
+                new Product {ProductID = 1, Name = "Lays", Price = 20},
+                new Product {ProductID = 2, Name = "Pringles", Price = 40}
+            };
+
+            Discount_1 discount1 = new Discount_1();
+
+            ValueCalculator calculator = new ValueCalculator();
+
+            ShoppingCart cart1 = new ShoppingCart(discount1, calculator) { Products = products };
+
+            var actualResult = cart1.CalculateTotal();
+            var expectedResult = 59.4;
+
+            Assert.AreEqual(expectedResult, actualResult);
```

Comment:
Add one more parameter here - message. text from this parameter will be displayed in case test fails at this line. For example, 'Total discount was calculated incorrectly'. 
Please, apply this recommendation to all your tests.